python/tchannel/tornado/dispatch.py

```diff
@@ -78,6 +76,13 @@ def handle_call(self, request, connection):
             request.endpoint += chunk
             chunk = yield request.argstreams[0].read()
 
+        # event: receive_request
+        if connection.tchannel:
+            request.tracing.name = request.endpoint
+            connection.tchannel.event_emitter.fire(
+                EventType.receive_request, request
```

Comment:
You shouldn't be storing a `TChannel` in `TornadoConnection`. Nor should you be accessing that `TChannel` inside the request handler through the `TornadoConnection`. Inject the necessary information (`EventEmitter` or `EventHook`) through other means instead of playing the 5-level-of-attribute-access game.